sstables/trie/bti_key_translation.cc

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include "bti_key_translation.hh"
+#include "sstables/mx/types.hh"
+
+namespace sstables::trie {
+
+static bytes_view bytespan_to_bytesview(std::span<const std::byte> s) {
+    return {reinterpret_cast<const bytes::value_type*>(s.data()), s.size()};
+}
+
+lazy_comparable_bytes_from_ring_position::lazy_comparable_bytes_from_ring_position(const schema& s, dht::ring_position_view rpv)
+    : _token(rpv.token())
+    , _pk(rpv.key() ? std::optional<partition_key>(*rpv.key()) : std::optional<partition_key>())
+    , _weight(rpv.weight())
```

Comment:
Note: with this, `lazy_comparable_bytes_from_ring_position` isn't going to handle the cursed ring position `dht::ring_position_view(dht::token::max(), nullptr, -1)` correctly.
For that position, `lazy_comparable_bytes_from_ring_position` will generate the same encoded bytes
as for `dht::ring_position_view(dht::token::last(), nullptr, -1)`, which is wrong.

(I deliberately didn't add any test for that case).

I'm only 80% sure `dht::ring_position_view(dht::token::max(), nullptr, -1)` isn't a thing that occurs anywhere, but maybe I shouldn't rely on that, and I should do:
```suggestion
    , _weight(_token.is_maximum() ? 1 : rpv.weight())
```